modules/core/shared/src/main/scala/gem/util/Zipper.scala

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2016-2018 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gem.util
+
+import cats._
+import cats.implicits._
+import cats.data.NonEmptyList
+
+/**
+ * A placeholder zipper for now. Stolen and modified from
+ * https://github.com/tpolecat/tuco/blob/master/modules/core/src/main/scala/tuco/util/zipper.scala
```

Comment:
I have this one as well:
https://github.com/gemini-hlsw/ocs3/blob/develop/modules/seqexec/web/shared/src/main/scala/seqexec/web/common/Zipper.scala

Reply:
I mostly just wanted a placeholder to keep going, but I did see that.  It has a lot of good stuff in there. I guess we could move it over and add `previous` and `next` which are missing.  I think the `Applicative` instance I wrote could be added as well.  I was mostly frightened off by the filter thing at the end which I didn't understand. Does it make sense move it out to a `yolo.syntax` package or something?

Reply:
On the other hand, if we don't need multiple manual options that I don't need a `Zipper` anyway.

Reply:
I cannot claim the Zipper on the seqexec is totally well done nor performant and it has some functions specific for the purpose I need
We could consolidate into one taking the nastier parts apart
One thing I think is good is to have a `Traverse[Zipper]` instance. that should produce an `Applicative` right?

Reply:
No, you need to define both `Traverse` and `Applicative`. And we should also define `Comonad`. And should contribute it to cats-collections.